src/natcap/invest/habitat_quality.py

```diff
@@ -895,13 +962,35 @@ def _compute_rarity_operation(
             code_index[code] = 0.0
 
     pygeoprocessing.reclassify_raster(
-        new_cover_path, code_index, rarity_path, gdal.GDT_Float32,
+        new_cover_path, code_index, rarity_raster_path, gdal.GDT_Float32,
         _OUT_NODATA)
 
+    _generate_rarity_csv(code_index, rarity_csv_path)
+
     LOGGER.info('Finished rarity computation on'
                 f' {os.path.basename(lulc_path_band[0])} land cover.')
 
 
+def _generate_rarity_csv(rarity_dict, target_csv_path):
+    """Generate CSV containing rarity values by LULC code.
+
+    Args:
+        rarity_dict (dict): dictionary containing LULC codes (as keys)
+            and their associated rarity values (as values).
+        target_csv_path (string): path to output CSV.
+
+    Returns:
+        None
+    """
+    lulc_codes = sorted(rarity_dict)
+    cols = ['LULC_code', 'rarity_value']
+    with open(target_csv_path, 'w') as csvfile:
+        csvfile.write(str(','.join(cols) + '\n'))
+        for lulc_code in lulc_codes:
+            row = [str(lulc_code), str(rarity_dict[lulc_code])]
+            csvfile.write(str(','.join(row) + '\n'))
```

Comment:
This looks like a very reasonable writing solution. But I did want to mention the `csv` module and `writer` object, https://docs.python.org/3/library/csv.html#csv.writer. One nice thing about using this is we don't have to worry about that newline character, which I think is best to leave up to the operating system when we can. Different OS's handle line endings differently. Not that this is the case here, but if we did have a list of rows we could also just write those out with `writerows(list_of_rows)`.

Reply:
I think I based this on an example I found in another model, but I will try switching over to `csv.writer`; thanks for the suggestion!